source/utilities.cc

```diff
@@ -1724,11 +1724,127 @@ namespace aspect
       return maximum_component_value[component];
     }
 
+
+
+    template <int dim>
+    void
+    AsciiDataLookup<dim>::reinit(const TableIndices<dim> &n_points_per_direction,
+                                 const std::vector<std::string> &column_names,
+                                 const std::vector<Table<dim,double> > &raw_data)
+    {
+      table_points = n_points_per_direction;
+      components = column_names.size();
+      data_component_names = column_names;
+      Assert(raw_data.size() == dim+components, ExcMessage("Incorrect number of columns specified."));
+
+
+      // compute maximum_component_value for each component:
+      maximum_component_value = std::vector<double>(components,-std::numeric_limits<double>::max());
+      for (unsigned int c=0; c<components; ++c)
+        {
+          unsigned int n_entries = 1;
+          for (unsigned int d=0; d<dim; ++d)
+            n_entries *= table_points[d];
+
+          for (unsigned int idx=0; idx<n_entries; ++idx)
+            maximum_component_value[c] = std::max(maximum_component_value[c], raw_data[dim+c](
+                                                    compute_table_indices(table_points, (components+dim)*idx)));
+        }
+
+      // In case the data is specified on a grid that is equidistant
+      // in each coordinate direction, we only need to store
+      // (besides the data) the number of intervals in each direction and
+      // the begin- and endpoints of the coordinates.
+      // In case the grid is not equidistant, we need to keep
+      // all the coordinates in each direction, which is more costly.
+      // Here we fill the data structures needed for both cases,
+      // and check whether the coordinates are equidistant or not.
+      // We also check the requirement that the coordinates are
+      // strictly ascending.
+
+      // The number of intervals in each direction
+      std::array<unsigned int,dim> table_intervals;
+
+      // Whether or not the grid is equidistant
+      coordinate_values_are_equidistant = true;
+
+      for (unsigned int i = 0; i < dim; i++)
```

Comment:
You can save yourself all of this if you used a better input format.

Reply:
Separately, it would be easier to read if you had chosen `d` as loop index over dimensions...